drivers/usb/udc/udc_stm32.c

```diff
@@ -962,13 +966,36 @@ static int priv_clock_enable(void)
 		return -ENODEV;
 	}
 
-#if defined(PWR_USBSCR_USB33SV) || defined(PWR_SVMCR_USV)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs) && defined(CONFIG_SOC_SERIES_STM32U5X)
+	/* Sequence to enable the power of the OTG HS on a stm32U5 serie : Enable VDDUSB */
+	if (__HAL_RCC_PWR_IS_CLK_DISABLED()) {
+		__HAL_RCC_PWR_CLK_ENABLE();
+		LL_PWR_EnableVddUSB();
+
+		/* Configure VOSR register of USB HSTranceiverSupply(); */
+		LL_PWR_EnableUSBPowerSupply();
```

Comment:
RM0456 mentions:
```
On STM32U59x/5Ax/5Fx/5Gx devices only:
a) Make sure the voltage scaling is in range 1 or in range 2 (using VOS[1:0]
in PWR_VOSR).
b) Make sure the EPOD booster clock is enabled (using PLL1MBOOST[3:0]
in RCC_PLL1CFGR).
```
Should we check them before setting USBPWREN and USBBOOSTEN in PWR_VOSR?

Reply:
The voltage scaling is set when configuring the stm32u5 sys clock. Yes, we can add a simple check on LL_PWR_REGU_VOLTAGE_SCALE1 or LL_PWR_REGU_VOLTAGE_SCALE2 and return an error if not correct.
The USB EPOD  booster is enabled by the LL_PWR_EnableUSBEPODBooster and yes we can add a loop waiting for the USBBOOSTRDY flag (LL_PWR_IsActiveFlag_USBBOOST )